exporter/exporterhelper/internal/queuebatch/persistent_queue.go

```diff
@@ -97,6 +102,14 @@ type persistentQueue[T any] struct {
 	stopped                  bool
 }
 
+// queueMetadata holds all persistent metadata for the queue.
+type queueMetadata struct {
```

Comment:
Let's extract this in a separate file please, already 400+ lines of code and 1000+ tests.